### PR TITLE
[MAP25] Compassrose

### PR DESCRIPTION
**Pull request by telemaxx**
grafik

### CODE REVIEW

**Review by wolfgang-ch — CHANGES_REQUESTED** [review]

**Comment by wolfgang-ch** [comment]
I'm wondering why this PR has 800 changes in Github but only a few in Eclipse. 

800 changes makes the website very slow in the browser when selecting/searching a modified file

image

image

**Comment by wolfgang-ch** [comment]
This PR is for `mytourbook:tm-map25` but it is not necessary any more that you update this/your branch because in Eclipse a PR can now be easily fetched from Github, just copy the url into the clipboard and select this action

image


image

File: bundles/net.tourbook/src/net/tourbook/map25/Map25View.java [hunk 1]
```diff
@@ -120,135 +135,167 @@ public class Map25View extends ViewPart implements
...
 
 // SET_FORMATTING_ON
+
+   private static final String           STATE_IS_LAYER_BASE_MAP_VISIBLE         = "STATE_IS_LAYER_BASE_MAP_VISIBLE";            //$NON-NLS-1$
+   private static final String           STATE_IS_LAYER_BOOKMARK_VISIBLE         = "STATE_IS_LAYER_BOOKMARK_VISIBLE";            //$NON-NLS-1$
+   private static final String           STATE_IS_LAYER_HILLSHADING_VISIBLE      = "STATE_IS_LAYER_HILLSHADING_VISIBLE";         //$NON-NLS-1$
+   private static final String           STATE_IS_LAYER_LEGEND_VISIBLE           = "STATE_IS_LAYER_LEGEND_VISIBLE";              //$NON-NLS-1$
+   private static final String           STATE_IS_LAYER_MARKER_VISIBLE           = "STATE_IS_LAYER_MARKER_VISIBLE";              //$NON-NLS-1$
+//   private static final String           STATE_IS_LAYER_OPEN_GL_TEST_VISIBLE     = "STATE_IS_LAYER_OPEN_GL_TEST_VISIBLE";        //$NON-NLS-1$
+   private static final String           STATE_IS_LAYER_COMPASS_ROSE_VISIBLE     = "STATE_IS_LAYER_COMPASS_ROSE_VISIBLE";        //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (telemaxx) [reply bundles/net.tourbook/src/net/tourbook/map25/Map25View.java:168]: done

File: bundles/net.tourbook/src/net/tourbook/map25/Map25App.java [hunk 2]
```diff
@@ -977,589 +1325,505 @@ private void saveState_MapPosition(final String suffixName) {
...
+         // force update
+         _currentOffOnline = null;
+
+         // this is necessarry that the new zoom level is applied, otherwise it do not work
+         setMapProvider(_selectedMapProvider);
+      }
    }
 
-   void stop() {
+   private void updateLayer_CompassRose() {
```
[the remark this answers is not in the document]
Comment (telemaxx) [reply bundles/net.tourbook/src/net/tourbook/map25/Map25App.java:1778]: removed

**Comment by telemaxx** [comment]
> I'm wondering why this PR has 800 changes in Github but only a few in Eclipse

yes, maybe i should have create a new fresh branch from main.
tm-map25 was old...

**Comment by telemaxx** [comment]
> but it is not necessary any more

hmm complicate, maybe due the weather my brain is overheating today.

next PR i try...

**Comment by wolfgang-ch** [comment]
> > but it is not necessary any more
> 
> hmm complicate, maybe due the weather my brain is overheating today.
> 
> next PR i try...

Just create the PR against the `main` branch but your branch has first to be merged with `main`, then there is only a difference between `main` and your branch

**Comment by wolfgang-ch** [comment]
After I've resetted `mytourbook:tm-map25` to the commit where you created the PR, now only 11 modified files are available which makes me happier than merging 800 files.

So you created the PR correctly but I did it not the old way.

In the future, do it simpler as described above https://github.com/mytourbook/mytourbook/pull/899#issuecomment-1192179158

image

File: bundles/net.tourbook/src/net/tourbook/map25/layer/marker/MarkerShape.java [hunk 3]
```diff
@@ -39,4 +41,9 @@ public enum MarkerShape {
     */
    ARROW,
 
+   /**
+    * using compass for a compass rose
+    */
+   COMPASS,
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/map25/layer/marker/MarkerShape.java:47]: The compassrose should be similar like in the 3D map

1. Position should be in the top right corner
2. It should not be so much tilt like the map

Position (1.)

image

Tilt (2.)

image

**Comment by wolfgang-ch** [comment]
Implemented in https://github.com/mytourbook/mytourbook/commit/5d4864665c5df31ba078c91fc72ba2e4d351dfd1

**Pull request closed without merging**